### PR TITLE
Topic functionality for user showcase

### PR DESCRIPTION
**Pull request by SEQUOIIA**
Adjusted Topic functionality to match what is described in the current task. The feature is still **very much** gated behind a feature flag.

### CODE REVIEW

File: src/Blaster.WebApi/Features/Capabilities/CapabilityServiceClient.cs [hunk 1]
```diff
@@ -80,27 +80,50 @@ public async Task DeleteCapability(string capabilityId)
...
             else if (response.StatusCode != HttpStatusCode.OK)
             {
-                throw new Exception($"Error! Capability was not updated in external service. Service returned ({response.StatusCode} - {response.ReasonPhrase})");
+                throw new Exception(
+                    $"Error! Capability was not updated in external service. Service returned ({response.StatusCode} - {response.ReasonPhrase})");
             }
+        }
+
+        public async Task SetCapabilityTopicCommonPrefix(string commonPrefix, string capabilityId)
+        {
```
Comment (kim-lindhard-dfds) [inline src/Blaster.WebApi/Features/Capabilities/CapabilityServiceClient.cs]: What is a commonPrefix?

File: src/Blaster.WebApi/Features/CapabilityDashboard/TopicComponent.js [hunk 2]
```diff
@@ -37,63 +37,37 @@ const TopicComponent = Vue.component("topic", {
         forwardNewMessageContract: function(description, type, schema, topicId) {
             this.$emit("messagecontractadd-new", description, type, schema, topicId);
             this.toggleShowAddMessageContract();
-        }
+        },
+        topicName: function(commonPrefix, topicMisc) {
```
Comment (kim-lindhard-dfds) [inline src/Blaster.WebApi/Features/CapabilityDashboard/TopicComponent.js]: is topicMisc the user input?

File: src/Blaster.WebApi/Features/CapabilityDashboard/TopicPrefixComponent.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        },
+
+        toSnakeCase: function(input) {
+            return input.replace(/\W+/g, " ")
+            .split(/ |\B(?=[A-Z])/)
+            .map(word => word.toLowerCase())
+            .join('_');
+        }
+    },
+    updated: function() {
```
Comment (kim-lindhard-dfds) [inline src/Blaster.WebApi/Features/CapabilityDashboard/TopicPrefixComponent.js]: Does this do anything?

File: fake_dependencies/tmp-up.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+docker-compose up --build
+docker-compose down
```
[the remark this answers is not in the document]
Comment (SEQUOIIA) [reply fake_dependencies/tmp-up.sh]: Helper script, wasn't supposed to be added.

File: src/Blaster.WebApi/Features/Capabilities/CapabilityServiceClient.cs (hunk 1 above)
Comment (SEQUOIIA) [reply src/Blaster.WebApi/Features/Capabilities/CapabilityServiceClient.cs]: Leftover from removal of unneeded functionality, to be removed.

File: src/Blaster.WebApi/Features/CapabilityDashboard/TopicPrefixComponent.js (hunk 3 above)
Comment (SEQUOIIA) [reply src/Blaster.WebApi/Features/CapabilityDashboard/TopicPrefixComponent.js]: Not anymore. It's a relic of the past that must be burned to the ground.

**Pull request merged**